### DISCUSSION TITLE
Subsequent property declarations must have the same type.  Property 'store' must be of type 'Store', but here has type 'Store'.

### DISCUSSION THREAD

**Discussion by JamesWilmot** [typed-ember/ember-cli-typescript#1565] [Q&A] [answered]
Having the following type clash with ```@types/ember-data``` and its subsequent import from ```@types/ember-data__adapter```. Any ideas?

```
node_modules/@types/ember-data/index.d.ts:2032:9 - error TS2717: Subsequent property declarations must have the same type.  Property 'store' must be of type 'Store', but here has type 'Store'.

2032         store: DS.Store;
             ~~~~~

  node_modules/@types/ember-data__adapter/node_modules/@types/ember-data/index.d.ts:2237:9
    2237         store: DS.Store;
                 ~~~~~
    'store' was also declared here.
```

```package.json``` excerpt:
```
    "@types/ember-data": "^4.4.11",
    "@types/ember-data__adapter": "^4.0.2",
    "@types/ember-data__model": "^4.0.1",
    "@types/ember-data__serializer": "^4.0.2",
    "@types/ember-data__store": "^4.0.3",
```

**Comment by chriskrycho** [comment, marked as answer]
This is a variation on the “Conflicting Type Dependencies” problem documented [here](https://docs.ember-cli-typescript.com/troubleshooting/conflicting-types)—so the solution is the same as in the docs: make sure you have only a single version of that type in your *resolved* dependency graph (not just top-level dependencies you specified, *any copy at all*). See the linked docs for more details!
